src/reconciliation.js

```diff
@@ -221,19 +221,29 @@ export function patchAccount(
     next.balance = BigNumber(updatedRaw.balance);
     changed = true;
   }
-
-  if (updatedRaw.balanceHistory) {
-    const { week } = updatedRaw.balanceHistory;
+  const { balanceHistory } = updatedRaw;
+  if (balanceHistory) {
+    const { week } = balanceHistory;
     const accountWeek = account.balanceHistory && account.balanceHistory.week;
-    const refreshBalanceHistory =
-      !week || // there is no week yet
-      !accountWeek || // there were no balance history yet
-      // last datapoint changes will force a refresh
-      week[week.length - 1][1] !==
-        accountWeek[accountWeek.length - 1].value.toString();
-    if (refreshBalanceHistory) {
-      next.balanceHistory = fromBalanceHistoryRawMap(updatedRaw.balanceHistory);
-      changed = true;
+    if (week && accountWeek) {
+      const [lastDateRaw, lastValue] = week[week.length - 1];
+      const { date: lastAccountDate, value: lastAccountValue } = accountWeek[
+        accountWeek.length - 1
+      ];
+      const lastDate = new Date(lastDateRaw);
+
+      const isSameDay =
+        lastDate.getFullYear() === lastAccountDate.getFullYear() &&
+        lastDate.getMonth() === lastAccountDate.getMonth() &&
+        lastDate.getDate() === lastAccountDate.getDate();
```

Comment:
Would it work with `lastDateRaw === lastAccountDate.toISOString()` ? (Hourly is coming soon)

Reply:
Solved in current code